plugins/foreman_remote_execution/0.1/index.md

```diff
@@ -0,0 +1,394 @@
+
+This plugin enables Foreman to run arbitrary commands on hosts using
+different providers. Initially only an SSH provider is supported but we
+plan to add more. Communication goes through the smart proxy so Foreman
+does not have to have direct access to the target host and can scale
+to control many hosts. A command can be customized similarly to
+provisioning templates or partition tables. A number of templates are included
+for running basic commands, installing packages, etc. For more details see the
+[Job
+Templates](plugins/foreman_remote_execution/{{page.version}}/index.html#3.1JobTemplates)
+section.
+
+Below is a short demo of running a single command on a particular host, with
+the ability to view the output live:
+
+![GIF Demo](/plugins/foreman_remote_execution/{{page.version}}/remote_execution.gif)
+
+The command can be executed on multiple servers at once while specific
+parts of it can vary per host - in other words you can use variables
+in your commands. Variable values can be filled by given host fact,
+smart class parameter, smart variable or even host parameter. Last but
+not least, like the "command" example above, the value can be specified by user
+when he or she triggers the execution.
+
+A few examples of tasks that this plugin helps you achieve:
+
+- install/update/remove package
+- bootstrap configuration management agent
+- trigger Puppet/Salt/Chef/... run
+
+## 1.1 Components
+
+The whole remote execution system is pluggable, making it easy to add
+more providers of communication later. Therefore it's split into
+several components that must be installed on Foreman  and Foreman proxy.
+
+<table class="table table-bordered table-condensed">
+  <tr>
+    <th>Plugin</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>foreman_remote_execution</td>
+    <td>Required, contains the UI part and currently also SSH provider, depends on foreman-tasks</td>
+  </tr>
+  <tr>
+    <td>foreman_tasks</td>
+    <td>Required, keep status of all executed commands, allows debugging if something goes wrong</td>
+  </tr>
+  <tr>
+    <td>smart_proxy_remote_execution_ssh</td>
+    <td>Smart proxy plugin, required for using SSH provider, proxies the communication between Foreman and hosts, requires smart_proxy_dynflow</td>
+  </tr>
+  <tr>
+    <td>smart_proxy_dynflow</td>
+    <td>Smart proxy plugin, runs the jobs asynchronously and keeps status of commands executed via this proxy. This plugins uses relational SQL DB (sqlite).</td>
+  </tr>
+</table>
+
+Each component is packaged for platforms that Foreman is officially
+packaged for. Since both Foreman and Smart proxy plugins requires
+Dynflow, Ruby 1.9+ is required.
+
+Foreman 1.10+ is compatible with 0.1.* versions of the plugin.
+
+# 2. Installation
+
+## 2.1 Plugin Installation
+
+<div class="alert alert-info">
+The Smart proxy component requires a system ruby of 1.9 or higher.  Currently, this means the proxy can't be installed on CentOS 6/RHEL 6 or Ubuntu 12.04.
```

Comment:
You'll probably find this in all of my theforeman.org commits too... an unbreakable habit.